### DISCUSSION TITLE
Are other services open source?

### DISCUSSION THREAD

**Discussion by asemandeveloper** [langgenius/dify#5086] [Suggestion]
Will other services of the project, such as web services and api, be open source?

**Comment by patryk20120** [comment]
Web and api are open source.

**Reply by asemandeveloper** [reply]
You did not understand what I meant
I know that web and API are open source. When will the rest of the project be open sourced?

**Reply by patryk20120** [reply]
Sorry :) There is only sandbox which will probably go to public when it will be mature enough, at least that what I heard. Do you need something specific out of it?

**Reply by asemandeveloper** [reply]
Yes. I want to work on AI workflow and implement the right ideas.

**Reply by patryk20120** [reply]
I think you dont need a sandbox for doing that.

**Reply by asemandeveloper** [reply]
very well. Which part of the project should I use and develop to work with AI Workflow and create new nodes?
Also give an explanation about sandbox.
